org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/ContentModelPlugin.java

```diff
@@ -131,33 +132,32 @@ private void updateSettings(ContentModelSettings settings, ISaveContext context)
 			boolean catalogPathsChanged = contentModelManager.setCatalogs(settings.getCatalogs());
 			if (catalogPathsChanged) {
 				// Validate all opened XML files
-				context.collectDocumentToValidate(d -> {
-					DOMDocument xml = context.getDocument(d.getDocumentURI());
-					if (xml == null) {
-						return false;
-					}
-					xml.resetGrammar();
-					return true;
-				});
+				validateAllOpenedDocument(context);
 			}
 		}
 		if (settings.getFileAssociations() != null) {
 			// Update XML file associations
 			boolean fileAssociationsChanged = contentModelManager.setFileAssociations(settings.getFileAssociations());
 			if (fileAssociationsChanged) {
 				// Validate all opened XML files
-				context.collectDocumentToValidate(d -> {
-					DOMDocument xml = context.getDocument(d.getDocumentURI());
-					xml.resetGrammar();
-					return true;
-				});
+				validateAllOpenedDocument(context);
 			}
 		}
 		// Update use cache, only if it is set in the settings.
 		Boolean useCache = settings.isUseCache();
 		if (useCache != null) {
 			contentModelManager.setUseCache(useCache);
 		}
+		// Download external resources
+		XMLDownloadExternalResourcesSettings downloadExternalResources = settings.getDownloadExternalResources();
+		boolean downloadExternalResourcesEnabled = downloadExternalResources == null
+				|| downloadExternalResources.isEnabled();
+		if (contentModelManager.isDownloadExternalResources() != downloadExternalResourcesEnabled) {
+			contentModelManager.setDownloadExternalResources(downloadExternalResourcesEnabled);
+			// Validate all opened XML files
+			validateAllOpenedDocument(context);

```

Comment:
if download is now disabled, why do we need to revalidate everything?

Reply:
When you have an XML which references a bad DTD, when you refresh the settings, the error message is different. So we need to revalidate.